### PR TITLE
Organize code into workspaces

### PR DESCRIPTION
**Pull request by amackillop**
This moves the protos into their own crate/workspace alongside the `cli` and `client` crates. Decided to keep the main server code in `src`. Will help prevent merge conflicts

### CODE REVIEW

**Review by saidalaoui-sq — APPROVED** [review]

File: common/build.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 fn main() -> Result<(), Box<dyn std::error::Error>> {
 	tonic_build::configure()
 		.protoc_arg("--experimental_allow_proto3_optional")
-		.protoc_arg("--proto_path=../src/proto")
+		.protoc_arg("--proto_path=../ldk-node-hack-server/src/proto")
```
[the remark this answers is not in the document]
Comment (amackillop) [reply common/build.rs:4]: Yeah we need to remove this. Tonic was used to generate the protos in common. We forgot last night that we didn't need a grpc client

File: common/Cargo.toml [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-edition = "2021"
+version.workspace = true
+edition.workspace = true
 
 [dependencies]
-prost = "0.12.6"
 reqwest = "0.12.5"
 tonic = "0.11.0"
+prost = { workspace = true }
+tokio = { workspace = true }
```
[the remark this answers is not in the document]
Comment (amackillop) [reply common/Cargo.toml:10]: Integration test code will use the client in common which is also used by the CLI

**Comment by amackillop** [comment]
I might need to rearrange this to not use the virtual manifest in the project root. Doesn't work well with the current docker setup for integration testing

File: common/Cargo.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (amackillop) [reply common/Cargo.toml:10]: Yeah could just be client and the generated protos

[the remark this answers is not in the document]
Comment (amackillop) [reply common/Cargo.toml:10]: Actually better to have the protos in their own crate

**Pull request merged**